c7n/resources/rdscluster.py

```diff
@@ -89,6 +90,62 @@ def process(self, resources, event=None):
 RDSCluster.filter_registry.register('network-location', net_filters.NetworkLocation)
 
 
+@RDSCluster.filter_registry.register('db-cluster-allowed-parameter')
```

Comment:
naming here is a bit odd, let's simplify to just `parameter` or `cluster-parameter`

Reply:
Updated to `db-cluster-parameter` for consistency with `modify-db-cluster`, does that work? Or do you want to cut it down to just `cluster-parameter`?

Reply:
its already namespaced under `rds-cluster` resource so preference would be to remove redundancy.

Reply:
Makes sense, got it - my worry is that this isn't filtering on the value stored in the parameter though, this is filtering on the `AllowedValues` field of a parameter, which is slightly different.

Thoughts on naming this something like `allowed-parameter`/`allowedvalues-parameter`/`allowed-parameter-vals`? Trying to avoid confusion if someone uses this thinking it's the same thing as RDS' `parameter` filter, though it's actually different

Reply:
Bumping this -- thoughts @kapilt ?

Reply:
it feels wierd just having this on allowed parameters rather just a parameter filter that can also poke at allowed values.

Reply:
Hi Kapil, Happy New Year! Apologies for the delayed response.

Looking at the existing db-parameter filter for rds, in https://github.com/cloud-custodian/cloud-custodian/blob/d79646d8d4f2d73b9e3d58ab1a006ce6cec0dd9e/c7n/resources/rds.py#L1553 we are pulling the resources. My code in this PR takes a similar approach; The results stored in `param_list` for a db cluster looks like the following:

`[{'ParameterName': 'test_name', 'ParameterValue': '0', 'Description': 'Test Description', 'Source': 'engine-default', 'ApplyType': 'static', 'DataType': 'boolean', 'AllowedValues': '0,1', 'IsModifiable': True, 'ApplyMethod': 'pending-reboot', 'SupportedEngineModes': ['provisioned']}, ...]`

In the next line of code, you can see that the existing rds `db-parameter` filter contains: `for p in param_list if 'ParameterValue' in p`, which hardcodes pulling out the `ParameterValue` field value. For the intent of this PR, we want to have a similar filter for db clusters that filters on `AllowedValues` rather than `ParameterValue`.

I based this PR off of the existing implementation in rds `db-parameter` that pulls the hardcoded `AllowedValues`. Should we look into opening up the entire functionality of `db-parameter` to allow specifying what value we want to filter on (i.e. can base off of `ParameterValue` or `AllowedValues`)?

Otherwise, if we want to keep these filters separate, I could also add a filter for DB Clusters `ParameterValue` to match rds `db-parameter` filter, in addition to this PR's filter for the `AllowedValues` field.

Reply:
Wanted to follow up on this and get some input @kapilt